service/rtc/server.go

```diff
@@ -147,25 +154,37 @@ func (s *Server) Stop() error {
 func (s *Server) msgReader() {
 	for msg := range s.sendCh {
 		if err := msg.IsValid(); err != nil {
-			s.log.Error("invalid message", mlog.Err(err))
+			s.log.Error("invalid message", mlog.Err(err), mlog.Int("msgType", int(msg.Type)))
+			continue
+		}
+
+		s.mut.RLock()
```

Comment:
I wonder if it would be better to lock the maps individually, instead of the entire struct? That might cause unnecessary locking for something that's going to happen quite often, right?

Reply:
It helps that this is read-locking which means it will only be waiting if at the same time a new session is being added which is a very fast operation to start with. The mutex per map optimization is totally valid but I am very cautious to add more complexity unless it's really needed. We will monitor everything of course, taking CPU profiles and such and if lock contention becomes an issue I am happy to improve further.